app/models/iiif_manifest_harvester.rb

```diff
@@ -16,9 +18,26 @@ def canvases
     manifest.sequences.first.canvases
   end
 
+  def ranges_for(canvas_id)
+    ranges.select do |range|
+      normalized_canvases = range.canvases.map { |c| c.partition('#').first }
```

Comment:
I _think_ we're doing the correct thing here in that if the canvas ID stored in the range has a reference to a xywh section we would still want it to match the given canvas ID that we're indexing.

Reply:
Yes, this is correct, for indexing purposes.